internal/pkg/pct/install_test.go

```diff
@@ -320,6 +330,107 @@ template:
   id: good-project
   author: puppetlabs
   version: 1.0.0
+`,
+				},
+			},
+		},
+		{
+			name: "if the git URI is invalid",
+			args: args{
+				gitUri: "invalid-uri",
+			},
+			expected: expected{
+				errorMsg: "Could not parse template uri",
+			},
+		},
+		{
+			name: "if the URI is valid but does not direct to a git repository",
+			args: args{
+				gitUri: "http://example.com/templates",
+			},
+			expected: expected{
+				errorMsg: "Could not clone git repository:",
+				filepath: "",
+			},
+			mockExecutions: mockExecutions{
+				name:        "git",
+				args:        []string{"clone", "http://example.com/templates", filepath.Join(tempWorkingPath, "temp")},
+				responseCmd: exec.Command("exit", "1"),
+			},
+		},
+		{
+			name: "if the URI is valid and does direct to a git repository",
+			args: args{
+				gitUri:    "https://github.com/puppetlabs/pct-test-template-01.git",
+				targetDir: templatePath,
+			},
+			expected: expected{
+				errorMsg: "",
+				filepath: filepath.Join(templatePath, "test-user", "test-template", "0.1.0"),
+			},
+			mockExecutions: mockExecutions{
+				name:        "git",
+				args:        []string{"clone", "https://github.com/puppetlabs/pct-test-template-01.git", filepath.Join(tempWorkingPath, "temp")},
+				responseCmd: exec.Command("echo", "test"),
+			},
+			mocks: mocks{
+				dirs: []string{
+					filepath.Join(tempWorkingPath, "temp"),
+				},
+				files: map[string]string{
+					filepath.Join(tempWorkingPath, "temp", "pct-config.yml"): `---
+template:
+  id: test-template
+  author: test-user
+  version: 0.1.0
+`,
+				},
+			},
+		},
+		{
+			name: "if the git repository does not contain a 'pct-config.yml' file",
+			args: args{
+				gitUri:    "https://github.com/puppetlabs/pct-test-template-01.git",
+				targetDir: templatePath,
+			},
+			expected: expected{
+				errorMsg: "Invalid config:",
+			},
+			mockExecutions: mockExecutions{
+				name:        "git",
+				args:        []string{"clone", "https://github.com/puppetlabs/pct-test-template-01.git", filepath.Join(tempWorkingPath, "temp")},
+				responseCmd: exec.Command("echo", "test"),
+			},
+			mocks: mocks{
+				dirs: []string{
+					filepath.Join(tempWorkingPath, "temp"),
+				},
+			},
+		},
+		{
+			name: "if git is not installed",
+			args: args{
+				gitUri:    "https://github.com/puppetlabs/pct-test-template-01.git",
+				targetDir: templatePath,
+			},
+			expected: expected{
+				errorMsg: "Could not clone git repository:",
```

Comment:
Just thinking we may want to have a better error message back if `git` is not installed, to direct the user to installing `git`?